### PR TITLE
create moderator feature

### PR DESCRIPTION
**Pull request by DMTRIP**

### CODE REVIEW

**Review by Viktor-Hordynskyi — APPROVED** [review]

**Review by Viktor-Hordynskyi — COMMENTED** [review]
Screenshot_3
you need to fix these problems

File: middlewares/route-validator.js [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
 const empty = 'cannot be empty';
 
+// yyyy-mm-dd
+const datePattern = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
+const isUnique = (field) => async (value) => {
+  const mod = await moderatorModel.findOne({ where: { [field]: value } });
+  if (mod) {
+    return Promise.reject();
+  }
+  return Promise.resolve();
```
Comment (innomaxx) [inline middlewares/route-validator.js:14]: `Promise.resolve` is useless. Async function always returns Promise\<T\>

File: services/admin-service.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const moderatorModel = require('../models/moderator.model');
+
+class AdminService {
+  async createModerator(data) {
```
Comment (innomaxx) [inline services/admin-service.js:4]: Useless async function  + try catch declaration
```
createModerator(data) {
    const ...
    const ...
    return moderatorModel.create({ .... })
}
```

File: middlewares/upload-images.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    cb(null, './uploads/');
+  },
+  filename(req, file, cb) {
+    cb(null, new Date().toISOString() + file.originalname);
+  },
+});
+
+const fileFilter = (req, file, cb) => {
+  // reject a file
+  if (file.mimetype === 'image/jpeg' || file.mimetype === 'image/png') {
```
Comment (innomaxx) [inline middlewares/upload-images.js]: Suggestion:
```
if (["image/jpeg", "image/png"].includes(file.mimetype))
```

File: middlewares/route-validator.js [hunk 4]
```diff
@@ -54,4 +65,38 @@ module.exports = {
...
+        .not().isEmpty().withMessage('Birth date is required')
+        .matches(datePattern)
+        .withMessage('Date must be yyyy-mm-dd format'),
+      check('email')
+        .not().isEmpty().withMessage('E-mail is required')
+        .isEmail()
+        .withMessage('email bad is bad formatted')
+        .custom(isUnique('email'))
+        .withMessage('E-mail must be unique'),
+      check('password', empty)
```
Comment (marrybbb) [inline middlewares/route-validator.js:91]: code duplicate  for check (): name, surname, password

File: middlewares/route-validator.js [hunk 5]
```diff
@@ -54,4 +65,38 @@ module.exports = {
...
+        .not().isEmpty()
+        .isString(),
+    ],
+    checkUniqueField: [
```
Comment (marrybbb) [inline middlewares/route-validator.js]: Better to write functions for such operations .
For example
function isEmpty(filed){ return check(field, empty).not().isEmpty(); } checUniqueFiled: [ isEmpty('filed'), isEmpty('value')]
1 - easier supporting
3 - DRY.

Comment (marrybbb) [inline middlewares/route-validator.js:95]: Better to write functions for such operations .
For example
`function isEmpty(filed){ return check(field, empty).not().isEmpty(); }
 checUniqueFiled: [ isEmpty('filed'), isEmpty('value')]`
1 - easier supporting
2 - DRY.

File: middlewares/route-validator.js (hunk 4 above)
Comment (DMTRIP) [reply middlewares/route-validator.js:91]: ???

Comment (marrybbb) [reply middlewares/route-validator.js:91]: you write the same code  3 times but in this case it is possible to remake logic and write it once.

check('name')
        .not().isEmpty()
        .isString(),
      check('surname')
        .not().isEmpty()
        .isString(),
and the same for password.

**Pull request closed without merging**